server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -923,11 +919,14 @@ public static ClusterState readFrom(StreamInput in, DiscoveryNode localNode) thr
         builder.routingTable = RoutingTable.readFrom(in);
         builder.nodes = DiscoveryNodes.readFrom(in, localNode);
         if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)) {
-            builder.transportVersions(in.readMap(TransportVersion::readVersion));
+            builder.versionsWrappers(in.readMap(streamInput -> new VersionsWrapper(TransportVersion.readVersion(streamInput))));
```

Comment:
Could we move the deserialization into `VersionsWrapper` (i.e. this should be `in.readMap(VersionsWrapper::readVersion)`?